editor/src/components/canvas/controls/new-canvas-controls.tsx

```diff
@@ -155,6 +159,45 @@ interface NewCanvasControlsClassProps {
   windowToCanvasPosition: (event: MouseEvent) => CanvasPositions
 }
 
+const selectElementsThatRespectLayout = createSelector(
```

Comment:
this is our new reselect selector! here's how it works: `createSelector` is a variadic function, the first n parameters are selectors each getting a part of the editor store, and the last parameter is a combiner which receives as arguments the values that were selected from the EditorStore.

The idea is that if the values returned by the individual selectors is `===` equal to the previously selected value, the reselect selector will return with the previously selected value instead of calling the combiner. Reselect allows us to combine selectors into larger selectors and in turn combine those into even larger selectors.

The big benefit this gives us is that if we previously had a Zustand selector that had to run an expensive computation, now we can put smaller selectors in front of it, just like in this example. If none of the navigatorTargets, propertyControlsInfo, openImports, openFilePath, rootComponents and jsxMetadataKILLME changes, we simply never call the combiner (which takes 3ms!) which saves us 3ms.

This memoized reselect selector takes 0.01ms to return if those parts of the state did not change, which is very promising as most of our current zustand selectors seem to eat around 0.05ms each.

Reply:
if the variadic style with the combiner being the _last_ parameter irks you, there's an alternative form:
`createSelector([reducer1, reducer2, ...], combiner)` where you put the reducers in an array. Has the same typescript support and everything